pages/components/display-product-modal.tsx

```diff
@@ -70,6 +68,25 @@ export default function DisplayProductModal({
     return [dateString, timeString];
   };
 
+  const handleShare = async () => {
+    // The content you want to share
+    const shareData = {
+      title: title,
+      url: `https://shopstr.store/listing/${productData.id}`,
+    };
+    // Check if the Web Share API is available
+    if (navigator.share) {
+      // Use the share API
+      await navigator.share(shareData);
+    } else {
+      // Fallback for browsers that do not support the Web Share API
+      navigator.clipboard.writeText(
+        `https://shopstr.store/listing/${productData.id}`,
+      );
+      alert("Listing URL copied to clipboard!");
```

Comment:
I think we should move away from alerts, it makes the site seem sorta unpolished. Perhaps an animation of copied to clipboard would be sufficient

Reply:
We have to change all the copied to clipboard uses then and make an animation of some sort. Best to be done in a separate pr I think.